http/http-advanced/src/main/java/io/quarkus/ts/http/advanced/SseResource.java

```diff
@@ -0,0 +1,90 @@
+package io.quarkus.ts.http.advanced;
+
+import java.util.Arrays;
+import java.util.concurrent.locks.LockSupport;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.client.WebTarget;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.sse.OutboundSseEvent;
+import jakarta.ws.rs.sse.Sse;
+import jakarta.ws.rs.sse.SseEventSink;
+import jakarta.ws.rs.sse.SseEventSource;
+
+@Path("/sse")
+public class SseResource {
+    @Context
+    Sse sse;
+
+    @GET
+    @Path("/client")
+    public String sseClient() {
+        try {
+            return consumeSse();
+        }
+        // in case that https://github.com/quarkusio/quarkus/issues/36402 throws java.lang.RuntimeException: java.lang.ClassNotFoundException:
+        // catch it and return the error message
+        catch (RuntimeException exception) {
+            return exception.getMessage();
+        }
+    }
+
+    private String consumeSse() {
+        StringBuilder response = new StringBuilder();
+
+        /*
+         * Client connects to itself (to server endpoint running on same app),
+         * because for https://github.com/quarkusio/quarkus/issues/36402 to reproduce client must run on native app.
+         * Which cannot be done in test code itself.
+         * This method acts just as a client
+         */
+        WebTarget target = ClientBuilder.newClient().target("http://localhost:" + getQuarkusPort() + "/api/sse/server");
+        SseEventSource updateSource = SseEventSource.target(target).build();
+        updateSource.register(ev -> {
+            response.append("event: ").append(ev.getName()).append(" ").append(ev.readData());
+            response.append("\n");
+
+        }, thr -> {
+            response.append("Error in SSE, message: ").append(thr.getMessage()).append("\n");
+            response.append(Arrays.toString(thr.getStackTrace()));
+        });
+        updateSource.open();
+
+        LockSupport.parkNanos(2_000_000_000L);
```

Comment:
same question here in regards of parking as below, thx

Reply:
SSE client (event) is asynchronous, so client needs to wait until the event really comes before return result. We can use other synchronization tool, but this one is simple and effective.

Reply:
ok